server/src/main/java/io/crate/execution/engine/aggregation/DocValueAggregator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.aggregation;
+
+import java.io.IOException;
+
+import javax.annotation.Nullable;
+
+import org.apache.lucene.index.LeafReader;
+
+public abstract class DocValueAggregator<T> {
+
+    public abstract T initialState();
```

Comment:
Once we use `DocValueAggregator` for grouping we should probably extend this for memory accounting. Not sure if already necessary for the global aggregate case, since there is only a single row.

Reply:
Right, I'll adapt it once I optimize that function (or any other where we allocate more than a couple of bytes)